### PR TITLE
feat(core): add `dataSource` field to MDL

### LINKED ISSUE #970: Add `dataSource` field in the MDL
**Is your feature request related to a problem? Please describe.**
Sometimes, we need to change our behavior according to the kind of data source. It's good to add a new field for it.

**Describe the solution you'd like**
- Add a new field in `Manifest`
```
pub data_source: Option<String>
```
- Implement the mapping function for the supported data source.
```rust
pub fn data_source_from_str(name: impl AsRef<str>) -> Option<String> {
    match name.to_lowercase().as_str() {
        "bigquery" => Some("bigquery"),
        ...
         _ => None,
    }
}
```
The return value couldn't be `String`. It may be an enum or something. Just write `String` temporarily.

### PR DESCRIPTION
**Pull request by goldmedal**
close #970

### CODE REVIEW

File: wren-core/core/src/mdl/manifest.rs [hunk 1]
```diff
@@ -18,6 +18,50 @@ pub struct Manifest {
...
+            DataSource::BigQuery => write!(f, "BIGQUERY"),
+            DataSource::Clickhouse => write!(f, "CLICKHOUSE"),
+            DataSource::Canner => write!(f, "CANNER"),
+            DataSource::Trino => write!(f, "TRINO"),
+            DataSource::MSSQL => write!(f, "MSSQL"),
+            DataSource::MySQL => write!(f, "MYSQL"),
+            DataSource::Postgres => write!(f, "POSTGRES"),
+            DataSource::Snowflake => write!(f, "SNOWFLAKE"),
+            DataSource::Datafusion => write!(f, "DATAFUSION"),
+            DataSource::DuckDB => write!(f, "DUCKDB"),
```
Comment (grieve54706) [inline wren-core/core/src/mdl/manifest.rs:62]: Curious, why make them uppercase?

**Review by grieve54706 — APPROVED** [review]

**Comment by grieve54706** [comment]
Could add a test case to assert this feature?

File: wren-core/core/src/mdl/manifest.rs (hunk 1 above)
Comment (goldmedal) [reply wren-core/core/src/mdl/manifest.rs:62]: No, specific reason. Just use the same style as the relationship type for the enum value.

**Comment by goldmedal** [comment]
> Could add a test case to assert this feature?

I added tests in `test_manifest_roundtrip` and the JSON file of the mdl to test if it can be serialized and deserialized well.

File: wren-core/core/src/mdl/manifest.rs (hunk 1 above)
Comment (grieve54706) [reply wren-core/core/src/mdl/manifest.rs:62]: The ibis-server and web use lowercase. Does this change will not affect the JSON format?

Comment (goldmedal) [reply wren-core/core/src/mdl/manifest.rs:62]: I think it won't affect it. We also accept the lowercase value.

Comment (grieve54706) [reply wren-core/core/src/mdl/manifest.rs:62]: When the ibis-server calls `ManifestExtractor` and then gets the object with an uppercase data source name, can it map to the `DataSource` enum of ibis-server?

Comment (goldmedal) [reply wren-core/core/src/mdl/manifest.rs:62]: It depends on how to implement it. By the way, why do we need to map to the enum of ibis-server?

Comment (goldmedal) [reply wren-core/core/src/mdl/manifest.rs:62]: It's fine for me to change them to lowercase if you think it's better. It doesn't matter actually. 🤔

Comment (goldmedal) [reply wren-core/core/src/mdl/manifest.rs:62]: Thanks

**Pull request merged**